app/pages/project/instances/instance/tabs/MetricsTab.tsx

```diff
@@ -54,27 +54,81 @@ function DiskMetric({
     { placeholderData: (x) => x }
   )
 
+  const isBytesChart = unit === 'Bytes'
+
+  let unitForSet = ''
+  let cycleCount = 0
+  let label
+  const divisorBase = isBytesChart ? 1024 : 1000
+  if (metrics?.items?.length) {
+    // Because we're using cumulative metrics, we can use the last (and therefore largest)
+    // value in the set to determine the unit and the divisor for the set.
+    const largestValue =
+      (metrics?.items?.[metrics.items.length - 1].datum.datum as Cumulativeint64).value || 0
+
+    // We'll need to divide each number in the set by a consistent multiple of 1024 (for Bytes) or 1000 (for Counts)
+    // Figure out what that multiple is:
+    let transformedValue = largestValue
+    while (transformedValue > divisorBase) {
+      transformedValue = transformedValue / divisorBase
+      cycleCount++
+    }
+
+    // Now that we know how many cycles of "divide by 1024 || 1000" to run through
+    // (via cylceCount), we can determine the proper unit for the set
+    if (isBytesChart) {
+      const byteUnits = ['Bytes', 'KiB', 'MiB', 'GiB', 'TiB']
+      unitForSet = byteUnits[cycleCount]
+      label = `(${unitForSet})`
+    } else {
+      label = '(COUNT)'
+    }
+  }
+
+  const divisor = divisorBase ^ cycleCount
```

Comment:
I think this is supposed to be `**`. `^` is bitwise XOR. I'm seeing values of 1001 and 1025 for `divisor`.

<img width="206" alt="image" src="https://github.com/oxidecomputer/console/assets/3612203/835e6be6-7704-4cbb-871e-191c18ccc4aa">